interface/src/app/tribe_client/tribe_client.js

```diff
@@ -0,0 +1,39 @@
+angular.module('adage.tribe_client', [
+    'adage.tribe_client.resource',
+    'adage.tribe_client.directives',
+])
+
+.factory( 'UserFactory', ['User', function( User ) {
+    var promise = false;
+    var user = null;
+
+    return {
+        getUser: function() { //only call this when the promise completes
+            return user;
+        },
+        getPromise: function() {
+            if (!promise) {
+                promise = User.get({}, function( data ) {
+                    if (data.meta.total_count !== 0) {
+                        user = data.objects[0];
+                    }
+                });
+            }
+            return promise;
+        },
+        resetPromise: function() { // reset the promise in case we need to check the user again
+            promise = false;
+        },
+        setUser: function( newUser ) { //set the user object to have properties from the passed user
+            var member = null;
```

Comment:
I'm not sure this line is adding anything. Is it? I believe the `for` loops below each create a local `member`, which will mask this one.